addons/edi/models/edi_gateway.py

```diff
@@ -216,6 +224,37 @@ def _compute_ssh_host_fingerprint(self):
             else:
                 gw.ssh_host_fingerprint = None
 
+
+    @api.depends("server", "port")
+    def compute_outgoing_server(self):
+        """Get the outgoing server from the smtp host name and port if supplied"""
+        MailServer = self.env["ir.mail_server"]
+        if self.server:
+            domain = [("smtp_host", "=", self.server)]
+            if self.port:
+                domain += [("smtp_port", "=", self.port)]
+            # Order the servers by priority as is default behaviour
+            server = MailServer.search(domain, order="sequence Asc", limit=1)
+            if server:
+                self.outgoing_server = server
+                _logger.info(
+                    ("Outgoing server for '%s' set to '%s' with address %s")
+                    % (self.name, server.name, self.server)
+                )
+            else:
+                _logger.warning(
+                    (
+                        "Outgoing server '%s' not found - will use the server"
```

Comment:
We are not using any server here as the `self.outgoing_server = False`. Should be consider changing the message?

Reply:
We don't set the server here, what will happen is it will try to use the server that is returned by MailSever.search([], order='sequence Asc', limit=1), i.e the server with the lowest priority. We could in the message display the default server, but I chose not to to avoid another search - although I don't think it will be too costly.